### DISCUSSION TITLE
Migrating from a MySQL backed database to SQLite?

### DISCUSSION THREAD

**Discussion by davidwinter** [WordPress/sqlite-database-integration#110] [Q&A]
Is there a recommended approach to migrating an existing MySQL backed site to use SQLite and this plugin? Installing the plugin currently installs a new SQLite database and states you start afresh.

**Comment by tassoman** [comment]
**_"A long time ago in a galaxy far, far away...."_** I used to migrate my old old blog from Drupal, back to WordPress. 
In both the situations I used the WXR feature, the first to extract data to Drupal, and some year after, exporting back to WP.
Today we have also [WPCLI helping](https://developer.wordpress.org/cli/commands/import/). 
So, I guess you can have two installations, extract data from mysql version into a static WXR file. Finally import it in the SQLite WP.
I guess can work somehow. 🤔

**Reply by Raboo** [reply]
The wp export WXR files do not include site configuration (options). So I guess more steps are required to get the configuration out of the MySQL into SQLite

**Reply by tassoman** [reply]
Good point. It depends by actual plugin's installation procedure. If it's migrating options table also, you could go on with your existing website, plugins included. Otherwise, if it's doing it from scratch, you're going to have a clean newborn WP installation.

In the first situation, I guess the only thing missing would be content, that's why I suggested exporting, then re-importing it by WRX feature. I'm not very sure about SQLite integration business model, so my words are pure theorical suppositions.

**Comment by taicv** [comment]
Currently I use **All-in-One WP Migration and Backup** or **Updraft Plus**
1. Create a backup
2. Install **sqlite-database-integration**, switch to SQLite 
3. Take Installation step, go to WP-Admin
4. Restore backup file.

I still searching solution for a large number of site or convert MySQL dump files (.sql) to SQLite. Any result from your research?

**Reply by tassoman** [reply]
I guess converting the MySQL dump to SQLite could be wrong, because it depends by the schema. SQLite fields are different (and less) than MySQL fields. But the backup-restore technique sounds good.
Do backup keep track of original PostID numbers?
